packages/wallet/backend/src/backfillTrxDetails.ts

```diff
@@ -0,0 +1,124 @@
+import { createContainer } from '@/createContainer'
+import { env } from '@/config/env'
+import { Account } from '@/account/model'
+import { Transaction } from '@/transaction/model'
+import { Model } from 'objection'
+import { transformBalance, urlToPaymentId } from '@/utils/helpers'
+
+async function backfillTrxDetails() {
+  const container = await createContainer(env)
+  const knex = container.resolve('knex')
+  Model.knex(knex)
+  const rafikiClient = container.resolve('rafikiClient')
+  const rafikiService = container.resolve('rafikiService')
+  const gateHubClient = container.resolve('gateHubClient')
+
+  async function getOutgoingPaymentSecondParty(paymentId: string) {
+    try {
+      const outgoingPayment =
+        await rafikiClient.getOutgoingPaymentById(paymentId)
+
+      return rafikiService.getOutgoingPaymentSecondPartyByIncomingPaymentId(
+        urlToPaymentId(outgoingPayment.receiver)
+      )
+    } catch (e) {
+      console.warn('Error on getting receiver wallet address', e)
+    }
+  }
+
+  async function getIncomingPaymentSecondParty(paymentId: string) {
+    return rafikiService.getIncomingPaymentSenders(paymentId)
+  }
+
+  async function backfillCardTrx(account: Account) {
```

Comment:
We probably need try-catch in this function when, otherwise if one of the backfill attempts fails the process will stop.